### ISSUE TITLE
Noise artifects when trying to reproduce TDM

### ISSUE THREAD

**Issue by terrencewayne** [Luo-Yihong/TDM#8]
Hello, I am trying to reproduce TDM on SD3 based on the training code, replacing EDM to flow matching. But I get severe noise artifacts and blurred images in results, while DMD2 plays well in the same iteration. It seems that the image has not been totally denoised. Have you met the problem? Do you have any suggestions on where there might be bugs in my implementation?

**Comment by Luo-Yihong** [comment]
Nice to see that you have solved the problem.

x0 prediction and v prediction both work well in our experiments. But in using x0 prediction, you should reweight the loss by snr.

**Comment by terrencewayne** [comment]
So the weight matters a lot. Thanks for your reply. I did not use snr weights in x0 prediction
